### PR TITLE
[SCM-868] fix gitexe cannot deduce relative path

### PR DESCRIPTION
**Pull request by basinilya**

### CODE REVIEW

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java [hunk 1]
```diff
@@ -215,19 +225,23 @@ public void testConsumerModifiedFileInComplexDirectorySetup() throws IOException
 	public void testConsumerModifiedFileInComplexDirectoryWithSpaces() throws IOException {
 
 		File dir = createTempDirectory();
-		File subdir = new File( dir.getAbsolutePath() + "/sub Directory déjà vu special/" );
+        URI relativeCWD = URI.create( "" );
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java]: check indentation

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java [hunk 2]
```diff
@@ -215,19 +225,23 @@ public void testConsumerModifiedFileInComplexDirectorySetup() throws IOException
...
+		File subdir = new File( dir, "sub Directory déjà vu special" );
 		subdir.mkdir();
 		FileUtils.write( new File( subdir, "project.xml" ), "data" );
 
-		List<ScmFile> changedFiles = getChangedFiles( "M  \"sub Directory déjà vu special/project.xml\"", subdir, dir.toURI() );
+        List<ScmFile> changedFiles =
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java]: check indentation

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java [hunk 3]
```diff
@@ -215,19 +225,23 @@ public void testConsumerModifiedFileInComplexDirectorySetup() throws IOException
...
+            getChangedFiles( "M  \"sub Directory déjà vu special/project.xml\"", dir, relativeCWD );
 
 		assertNotNull( changedFiles );
 		assertEquals( 1, changedFiles.size() );
         assertEquals( "sub Directory déjà vu special/project.xml", changedFiles.get( 0 ).getPath() );
 
         FileUtils.write( new File( subdir, "test file with spaces and a déjà vu character.xml" ), "data" );
 
-		changedFiles = getChangedFiles( "M  \"sub Directory déjà vu special/test file with spaces and a déjà vu character.xml\"", subdir, dir.toURI() );
+        changedFiles =
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java]: check indentation

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java [hunk 4]
```diff
@@ -187,23 +187,33 @@ public void testConsumerModifiedFile()
...
+        URI u = GitStatusConsumer.uriFromPath( path );
+        assertEquals( path, u.getPath() );
+    }
+
 	// SCM-740
 	public void testConsumerModifiedFileInComplexDirectorySetup() throws IOException {
 
 		File dir = createTempDirectory();
-		File subdir = new File( dir.getAbsolutePath() + "/subDirectory/" );
+        URI relativeCWD = URI.create( "" );
```
[the remark this answers is not in the document]
Comment (basinilya) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumerTest.java]: checked.

**Comment by jira-importer** [comment]
Resolve #1097

**Comment by jira-importer** [comment]
Resolve #1097

**Comment by jira-importer** [comment]
Resolve #1155

**Pull request closed without merging**